src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/bfaction/BfActionReport.java

```diff
@@ -0,0 +1,93 @@
+        BfActionReportData bfActionReportData = new BfActionReportData();
+        var caseData = listingDetails.getCaseData();
+        if (!CollectionUtils.isEmpty(submitEvents)) {
+            List<BFDateTypeItem> bfDateTypeItems = new ArrayList<>();
+            for (SubmitEvent submitEvent : submitEvents) {
+                addBfDateTypeItems(submitEvent, caseData, bfDateTypeItems);
+            }
+            bfDateTypeItems.sort(new BFDateTypeItemComparator());
+            bfActionReportData.setBfDateCollection(bfDateTypeItems);
+        }
+
+        String caseTypeId = listingDetails.getCaseTypeId();
+        var office = UtilHelper.getListingCaseTypeId(caseTypeId);
+        bfActionReportData.clearReportFields();
+        bfActionReportData.setReportType(BROUGHT_FORWARD_REPORT);
+        bfActionReportData.setDocumentName(BROUGHT_FORWARD_REPORT);
+        bfActionReportData.setOffice(office);
+        bfActionReportData.setListingDate(caseData.getListingDate());
+        bfActionReportData.setListingDateFrom(caseData.getListingDateFrom());
+        bfActionReportData.setListingDateTo(caseData.getListingDateTo());
+        return bfActionReportData;
+    }
+
+    private void addBfDateTypeItems(SubmitEvent submitEvent, ListingData listingData,
+        List<BFDateTypeItem> bfDateTypeItems) {
+        if (!CollectionUtils.isEmpty(submitEvent.getCaseData().getBfActions())) {
+            for (var bfActionTypeItem : submitEvent.getCaseData().getBfActions()) {
+                var bfDateTypeItem = getBFDateTypeItem(bfActionTypeItem, listingData,
+                    submitEvent.getCaseData());
+                if (bfDateTypeItem != null && bfDateTypeItem.getValue() != null) {
+                    bfDateTypeItems.add(bfDateTypeItem);
+                }
+            }
+        }
+    }
+
+    private BFDateTypeItem getBFDateTypeItem(BFActionTypeItem bfActionTypeItem,
+                                                    ListingData listingData, CaseData caseData) {
+        var bfActionType = bfActionTypeItem.getValue();
+        if (!isNullOrEmpty(bfActionType.getBfDate()) && isNullOrEmpty(bfActionType.getCleared())) {
+            var bfDate = ReportHelper.getFormattedLocalDate(bfActionType.getBfDate());
+            boolean isValidBfDate = ReportHelper.validateMatchingDate(listingData, bfDate);
+
+            if (isValidBfDate) {
+                return createBFDateTypeItem(bfActionTypeItem, bfDate, caseData.getEthosCaseReference());
+            }
+        }
+        return null;
+    }
+
+    private BFDateTypeItem createBFDateTypeItem(BFActionTypeItem bfActionTypeItem, String bfDate,
+                                             String ethosCaseReference) {
+        var bfActionType = bfActionTypeItem.getValue();
+        var bfDateType = new BFDateType();
+        bfDateType.setCaseReference(ethosCaseReference);
+
+        if (!Strings.isNullOrEmpty(bfActionType.getAllActions())) {
```

Comment:
don't need to declare Strings as you've imported isNullOrEmpty

Reply:
sure